### PR TITLE
feat: Add Mentions component

### LINKED ISSUE #14542: Daparated Mention integration with draft.js
Add some warning messages when developers use `ContentState` instance from draft.js, being prepared for completely removing draft.js and immuatable.js in future.

### PR DESCRIPTION
**Pull request by zombieJ**
# TODO: WIP

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

close #14542

### 💡 Solution

Use `rc-mentions`

### 📝 Changelog

New component Mentions. Origin Mention component is now deprecated.
新增 Mentions 组件。原 Mention 组件被废弃。

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/mentions/demo/basic.md](https://github.com/ant-design/ant-design/blob/mentions/components/mentions/demo/basic.md)
[View rendered components/mentions/index.en-US.md](https://github.com/ant-design/ant-design/blob/mentions/components/mentions/index.en-US.md)
[View rendered components/mentions/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/mentions/components/mentions/index.zh-CN.md)

### CODE REVIEW

**Comment by afc163** [comment]
This pull request **introduces 1 alert** when merging ecb39c49b2ce44b152c2020fd68fcc3e7825436a into e88589d9eff4428c4d5dffa305c882141eef7ef6 - [view on LGTM.com](https://lgtm.com/projects/g/ant-design/ant-design/rev/pr-40c118239b5c511164508d1f87c9cc7e71b4b28f)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by ztplz** [comment]
New component？

**Comment by zombieJ** [comment]
> New component？

Yes. To replace of Mention (#14542)

File: components/mentions/index.zh-CN.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ type: 数据录入
 title: Mentions
 ---
 
-提及组件。
+3.19.0 中新增，提及组件。如果想查阅废弃 Mention 组件 api，请点击[这里](/components/mention)。
```
Comment (afc163) [inline components/mentions/index.zh-CN.md]: 最好不要修改组件本身的描述，放在下面。

Comment (afc163) [reply components/mentions/index.zh-CN.md]: 这句话看着不正常，读不通。

File: components/mentions/style/index.less [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    font-stretch: inherit;
+    line-height: inherit;
+    direction: inherit;
+    letter-spacing: inherit;
+    white-space: inherit;
+    text-align: inherit;
+    vertical-align: top;
+    word-wrap: break-word;
+    word-break: inherit;
+    tab-size: inherit;
```
Comment (afc163) [inline components/mentions/style/index.less]: 这一堆 inherit 是必要的么？

File: components/mention/index.zh-CN.md [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
 ---
 
-提及组件。
+提及组件。已废弃，请使用 [Mentions](/components/mentions) 代替。
 
-## 何时使用
+## 为何废弃？
 
-用于在输入中提及某人或某事，常用于发布、聊天或评论功能。
+Mention 组件使用了 [Draft.js](https://www.npmjs.com/package/draft-js) 进行提示定位，占用了约 11.6% 的包大小。因而我们决定使用更轻量级的解决方案以便于在未来降低整个包的大小。
```
Comment (afc163) [inline components/mention/index.zh-CN.md]: 搞个 Alert 高亮一下？

**Comment by afc163** [comment]
image

在 Form 里使用时，和报错信息的间距有点远了。

File: components/mentions/style/index.less (hunk 2 above)
Comment (zombieJ) [reply components/mentions/style/index.less]: 嗯，需要让 textarea 和 measure 保持一致才能使得渲染出来的位置相同。

**Comment by zombieJ** [comment]
Updated. Pls help to check again.

**Review by afc163 — APPROVED** [review]

**Pull request merged**